src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs

```diff
@@ -350,9 +351,22 @@ private void InnerMeasureCore(Size availableSize)
 				desiredSize.Height = LayoutRound(desiredSize.Height);
 			}
 
+			var oldWidth = this.GetActualWidth();
+			var oldHeight = this.GetActualHeight();
+
 			// DesiredSize must include margins
 			LayoutInformation.SetDesiredSize(this, desiredSize);
 
+			var newWidth = this.GetActualWidth();
+			var newHeight = this.GetActualHeight();
+
+			if (oldWidth != newWidth || oldHeight != newHeight)
+			{
+				// This is HACKY, and is specific to TextBlock which overrides GetActualWidth and GetActualHeight to return DesiredSize.
```

Comment:
@dr1rrb We'are comparing `Actual[Width|Height]` after a SetDesiredSize call.

For all elements, DesiredSize isn't going to change `Actual[Width|Height]`. However, for `TextBlock`, it's the case because of:

https://github.com/unoplatform/uno/blob/30014dd5103817fc401eff20a913cb4c3cba4b22/src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs#L1216-L1217

Reply:
Ok ... but why we need to raise the event twice? Is it really expected?

Reply:
It shouldn't be raising it twice, but thinking more, it will be raising it once in a case where it shouldn't have been raised.

It looks like currently we don't have any good event to listen for here:

https://github.com/unoplatform/uno/blob/519c422535527d70deda85e391b3a70b360c9481/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.skia.cs#L202

Will investigate more next week

Reply:
Fixed